hermeto/core/scm.py

```diff
@@ -209,3 +240,88 @@ def _reset_git_head(repo: Repo, ref: str) -> None:
             "Failed on checking out the Git repository. Please verify the supplied reference "
             f'of "{ref}" is valid.'
         )
+
+
+class GitSupportConfig:
+    """Base configuration class for package manager git support."""
+
+    def __init__(
+        self,
+        supported_schemes: list[str],
+        package_manager: PackageManagerType,
+    ):
+        """Initialize GitSupportConfig.
+
+        Args:
+            supported_schemes: List of supported git URL schemes (e.g. ['git+https', 'git+ssh'])
+            package_manager: The package manager this configuration is for
+        """
+        self.supported_schemes = supported_schemes
+        self.package_manager = package_manager
+
+
+def validate_git_dependency(url: str, config: GitSupportConfig) -> None:
```

Comment:
I don't think we need this validation. This merely validates URLs, more specifically - URL schemes. Instead, this should be done on the pydantic model validation level and that's what we already do.
However, looking at our existing `AcceptedUrl = Annotated[` definition for bundler, there's a bug we need to fix as it only assumes the `https` URL scheme for git dependency URLs.